Common/Data/Consolidators/RangeConsolidator.cs

```diff
@@ -0,0 +1,223 @@
+/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using Python.Runtime;
+using QuantConnect.Data.Market;
+using QuantConnect.Securities;
+using System;
+
+namespace QuantConnect.Data.Consolidators
+{
+    /// <summary>
+    /// This consolidator can transform a stream of <see cref="IBaseData"/> instances into a stream of <see cref="RangeBar"/>
+    /// </summary>
+    public class RangeConsolidator : BaseTimelessConsolidator
+    {
+        private bool _firstTick;
+        private decimal _minimumPriceVariation;
+        protected RangeBar CurrentRangeBar;
```

Comment:
Maybe could directly use `CurrentBar`

Reply:
Yeah, in the beginning I wanted to use that property instead of creating new field, but since `CurrentBar` is of type `IBaseData` It doesn't have `Volume` so I cannot use it in the inner overridden methods of `ClassicRangeConsolidator` and `RangeConsolidator`. On the other hand, though I could have used a private variable named `_currentBar`, it wouldn't have worked since I also needed to access that variable from a child class called `ClassicRangeConsolidator` and if it was protected, it's name should have been in PascalCase, therefore being equal to the property.